File: dynamic_listings/python/default/computer_science/redundancy_2.py
class List:
    node_list: list[Node] = None
    length: int = None

    # ...

    def __len__(self) -> int:
        return self.length

    def add_item(self, node: Node) -> None:
        # ... Normal operation ...
        # ...
        # We update our length counter
        self.length += 1

    def remove_item(self, node: Node) -> None:
        # ... Normal removal operation ...
        # ...
        # We update our length counter
        self.length -= 1

    def clear(self) -> None:
        # ... Normal clear operation ...
        # ...
        # We clear the length too
        self.length = 0

    # ...

File: dynamic_listings/rust/default/computer_science/redundancy_2.rs
struct List {
    node_list: Vec<Node>,
    length: usize,
    // ...
}

impl List {
    pub fn length(&self) -> usize {
        self.length
    }

    pub fn add_item(&mut self, node: &Node) {
        // ... Normal operation ...
        // ...
        // We update our length counter
        self.length = self.length + 1;
    }

    pub fn remove_item(&mut self, node: &Node) {
        // ... Normal removal operation ...
        // ...
        // We update our length counter
        self.length = self.length - 1;
    }

    pub fn clear(&mut self) {
        // ... Normal clear operation ...
        // ...
        // We clear the length too
        self.length = 0;
    }
    // ...
}
